ui/panels-plugin/src/plugins/bar-chart/bar-chart-model.ts

```diff
@@ -0,0 +1,49 @@
+// Copyright 2023 The Perses Authors
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import { ModeOption, SortOption } from '@perses-dev/components';
+import { CalculationType, Definition, UnitOptions } from '@perses-dev/core';
+import { OptionsEditorProps } from '@perses-dev/plugin-system';
+
+export const DEFAULT_UNIT: UnitOptions = { kind: 'Decimal' };
+
+/**
+ * The schema for a BarChart panel.
+ */
+export interface BarChartDefinition extends Definition<BarChartOptions> {
+  kind: 'BarChart';
+}
+
+/**
+ * The Options object type supported by the BarChart panel plugin.
+ */
+export interface BarChartOptions {
+  calculation: CalculationType;
+  unit: UnitOptions;
+  sort: SortOption;
```

Comment:
I think sort can also be optional and we can make 'asc' as default

Reply:
In general, i think we want as many defaults as possible so we don't have to explicitly define every option in the json schema

Reply:
Makes sense, I updated `unit`, `sort` and `mode` to be optional! ef0a2538ce1f30459e24f92dfd1f01ef2c9207de

Reply:
I'm getting some build errors, I think you also need to update the props in ModeSelector and SortSelector

Reply:
Oops, should be good now

Reply:
Not necessary for this PR, but I like the 'Reset to Defaults' button for this too to make it easy to revert to default spec, it should really be handled by the framework in the future and on StatChart and GaugeChart too
<img width="306" alt="image" src="https://github.com/perses/perses/assets/9369625/1782a1ee-9688-4174-bd5b-9e6c320fcf1d">